src/main/java/com/google/finapp/SpannerDaoJavaImpl.java

```diff
@@ -22,16 +22,17 @@
 
 import java.math.BigDecimal;
 
-final class SpannerDao {
+final class SpannerDaoJavaImpl implements SpannerDaoInterface {
```

Comment:
I think `SpannerDaoImpl` (and later `SpannerDaoJDBCImpl`) would be fine too, up to you.